node_modules/gh-auth/tests/test-oauth.js

```diff
@@ -0,0 +1,764 @@
+                                                                    AuthTestsUtil.assertDeleteOauthAccessTokenByToken(userAppClient, token2, function() {
+
+                                                                        // Ensure the second bearer client is non-functional
+                                                                        UsersTestsUtil.assertGetMe(bearerClient2, function(meAfterDelete2) {
+                                                                            assert.ok(meAfterDelete2.anon);
+                                                                            assert.ok(!meAfterDelete2.id);
+                                                                            return callback();
+                                                                        });
+                                                                    });
+                                                                });
+                                                            });
+                                                        });
+                                                    });
+                                                });
+                                            });
+                                        });
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        /**
+         * Test that verifies an access token for a disabled client cannot be used
+         */
+        it('verify access token for disabled client cannot be used', function(callback) {
+            AuthTestsUtil.assertCreateOauthClient(userAppAdminClient, TestsUtil.generateString(), REDIRECT_URI_VALID, null, function(oauthClient) {
+                AuthTestsUtil.assertGrantOauthAuthCode(userAppAdminClient, oauthClient.id, null, function(code) {
+                    AuthTestsUtil.assertExchangeOauthAuthCode(userAppAdminClient, oauthClient.id, oauthClient.secret, code, oauthClient.redirectUri, null, function(bearerClient, token) {
+
+                        // Disable the client with which the token is associated
+                        AuthTestsUtil.assertDisableOauthClient(userAppAdminClient, oauthClient.id, function(oauthClientDisabled) {
+
+                            // Ensure the bearer client is now anonymous
+                            UsersTestsUtil.assertGetMe(bearerClient, function(bearerMeAfterDisable) {
+                                assert.ok(bearerMeAfterDisable.anon);
+                                assert.ok(!bearerMeAfterDisable.id);
+                                return callback();
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        it('verify listing access tokens for a user', function(callback) {
```

Comment:
Can you also add a test where an app admin grants a user access to a client and then verifies the user can get the access token from the API ?